arm-software/linux/docs/PortingFromACfL.md

```diff
@@ -0,0 +1,101 @@
+## ACfL to ATfL Porting Guide
+
+This document outlines the key differences between using the Arm Toolchain for
+Linux (ATfL) and the Arm Compiler for Linux (ACfL). It also lists command-line
+options and macros that can be used with ATfL as alternatives to those used in
+ACfL.
+
+Both toolchains provide frontends for compiling C, C++, and Fortran code, and
+the names of these frontends are consistent across both: `armclang` for C,
+`armclang++` for C++, and `armflang` for Fortran.
+
+### Reference Version
+
+|Compiler                      |Version|
+|------------------------------|-------|
+|Arm Compiler for Linux (ACfL) |24.10  |
+|Arm Toolchain for Linux (ATfL)|20.0   |
+
+### ArmPL integration
+
+In ACfL, Arm Performance Libraries (ArmPL) are bundled with the compiler. To
+simplify usage, the `-armpl` flag is provided to automatically include the
+necessary headers and libraries for BLAS, LAPACK, FFT, and other numerical
+routines.
+
+In contrast, ATfL does not include ArmPL directly, but it depends on the ArmPL
+package, which is installed alongside the toolchain. To use ArmPL with ATfL, the
+users must manually specify the locations of the headers and libraries using
+`pkg-config`. Detailed instructions are available in the
+[Getting Started](./GettingStarted.md) guide.
+
+#### Note
+
+Vector math functions from libamath are accessible without manually specifying
+the libraries.
+
+### C/C++ Frontend
+
+The C/C++ frontend in ATfL is identical to the one used in ACfL. Both are built
+on the Clang frontend from the upstream LLVM project (`llvm-project/clang`).
+
+### Fortran Frontend
+
+The Fortran frontend in ATfL is based on the new LLVM Flang project
+(`llvm-project/flang`). This frontend is a modern, from-scratch implementation.
+In contrast, ACfL used the older Classic Flang frontend, available at
+[https://github.com/flang-compiler/flang](https://github.com/flang-compiler/flang).
+Because ATfL introduces a new Fortran frontend, this document will primarily
+focus on the differences in Fortran support between the two toolchains.
+
```

Comment:
When these are merged, it might be worth linking to the new documentation for flang's feature support:

- https://github.com/llvm/llvm-project/pull/132707
- https://github.com/llvm/llvm-project/pull/132195

(To be done after the ATfL 20 release because there is no LLVM-20 version of these documents)

Reply:
That's one possibility. Other was to copy those .md files (when they finally get merged) into our rpms/debs next to those .md files. I guess the links would be much easier to introduce. Nevertheless, this can only happen after these are merged, so it may remain not covered by this particular commit.